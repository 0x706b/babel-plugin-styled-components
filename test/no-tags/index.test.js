import { transform } from 'babel-core'
import * as fs from 'fs'

jest.mock('styled-components/package.json', () => ({ version: '4.0.0' }))

it('should swap out the import if styled-components v4+ is detected', () => {
  const fixture = transform(
    fs.readFileSync(__dirname + '/example.js', 'utf8'),
    { plugins: ['./src'] }
  ).code
  expect(fixture).toMatchSnapshot()
})
