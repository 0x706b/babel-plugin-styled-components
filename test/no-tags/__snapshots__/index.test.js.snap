// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should swap out the import if styled-components v4+ is detected 1`] = `
"\\"use strict\\";

var _styledComponents = _interopRequireDefault(require(\\"styled-components\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Named = _styledComponents.default.div\`
  width: 100%;
\`;
var NamedWithInterpolation = _styledComponents.default.div\`
  color: \${function (props) {
  return props.color;
}};
\`;
var Wrapped = (0, _styledComponents.default)(Inner)\`
  color: red;
\`;"
`;
